analyses/hello-clusters/01_perform-evaluate-clustering.Rmd

```diff
@@ -49,32 +62,443 @@ data_dir <- file.path(repository_base, "data", "current")
+
+### Evaluating ScPCA clusters
+
+ScPCA cell metadata already contains a column called `cluster` with results from an automated clustering. 
+These clusters were calculated using `bluster`, the same tool that `rOpenScPCA` uses, with the following parameters:
+
+* The Louvain algorithm
+* Jaccard weighting
+* 20 nearest neighbors
+
+In this example, we'll show how to use the cluster evaluation functions on these clusters.
+
+To begin, we'll prepare a data frame with two columns: `cell_id` containing cell barcodes, and `cluster` containing the cluster identities.
+
+```{r prepare scpca data frame}
+scpca_cluster_df <- data.frame(
+  cell_id = colnames(sce),
+  cluster = colData(sce)$cluster
+)
+
+scpca_cluster_df
+```
+
+We can run evaluation functions using this data frame and the PCA matrix from the object.
+
+```{r scpca silhouette}
+# Calculate silhouette width
+scpca_silhouette_df <- calculate_silhouette(pca_matrix, scpca_cluster_df)
+scpca_silhouette_df
+```
+
+```{r scpca purity}
+# Calculate neighborhood purity
+scpca_purity_df <- calculate_purity(pca_matrix, scpca_cluster_df)
+scpca_purity_df
+```
+
+When running `calculate_stability()`, we'll specify the same parameters originally used to build the clusters.
+We only recommend evaluating clusters with `calculate_stability()` if you know the original parameters used.
+
+```{r scpca stability}
+scpca_stability_df <- calculate_stability(
+  pca_matrix,
+  scpca_cluster_df,
+  # provide ScPCA clustering parameters
+  algorithm = "louvain",
+  weighting = "jaccard",
+  nn = 20
+)
+scpca_stability_df
+```
+
+
+
+## Working with objects directly
+
+As presented above, `rOpenScPCA` clustering functions take a PCA matrix with row names representing unique cell ids as their first argument.
+
+Instead of a matrix, you can alternatively pass in an SCE or Seurat object that contains a matrix.
+We show examples of this below with `rOpenScPCA::calculate_clusters()`, but this will also work for any of the evaluation functions as well:
+
+```{r run on sce}
+# Calculate clusters from an SCE object using default parameters
+cluster_results_df <- calculate_clusters(sce)
+cluster_results_df
+```
+
+
+```{r run on seurat}
+# Calculate clusters from a Seurat object using default parameters
+cluster_results_df <- calculate_clusters(seurat_obj)
+cluster_results_df
+```
+
+
+`rOpenScPCA` assumes that the PCA matrix is named `PCA` in SCE objects, and `pca` in Seurat objects (as they were in the above examples).
+If the PCA matrix you want to use in the object has a different name, you can provide the argument `pc_name`.
+We can see this below with an SCE object, for example:
+
+```{r use pc_name}
+# First, we'll rename the PCA matrix for demonstration
+reducedDimNames(sce) <- c("PCA_matrix", "UMAP")
+reducedDimNames(sce)
+
+# Calculate clusters from an SCE object using default parameters
+cluster_results_df <- calculate_clusters(
+  sce,
+  pc_name = "PCA_matrix"
+)
+cluster_results_df
+```
```

Comment:
This feels like a detail we don't need to demonstrate. I would remove this section, or at least the code for it.